### ISSUE TITLE
**TimeoutException: Prompt not found: the code is taking too long to finish or there is a syntax error. Last 1000 bytes read:**?

### ISSUE THREAD

**Issue by mbazan23** [byexamples/byexample#74]
**TimeoutException: Prompt not found: the code is taking too long to finish or there is a syntax error.
Last 1000 bytes read:**

I do not think it's a bug, it's just a query of an error:
1. I want to execute this example 'product_configuration.md'

2. Then I run ``byexample`` like this 'byexample -v @test/boptions bin/product_configuration.md.'
3. See error

**Expected behavior**
...
File 'bin/product_configuration.md': 3 examples [Ruby Prompt Finder]
File 'bin/product_configuration.md': 4 examples [Shell Prompt Finder]
File 'bin/product_configuration.md': 17 examples [Dshell Prompt Finder]
Initializing 3 runners...
 - Shell Runner
 - Ruby Runner
 - Dshell Runner
                                                                                                                                                                                                                                                                               
File bin/product_configuration.md, 24/24 test ran in 11.23 seconds                                                                                                                                                                                                             
[PASS] Pass: 24 Fail: 0 Skip: 0
Shutting down 3 runners...                                                                                                                                                                                                                                                     
 - Shell Runner
 - Ruby Runner
 - Dshell Runner

**Actual behavior**
File 'bin/product_configuration.md': 3 examples [Ruby Prompt Finder]
File 'bin/product_configuration.md': 4 examples [Shell Prompt Finder]
File 'bin/product_configuration.md': 17 examples [Dshell Prompt Finder]
Initializing 3 runners...
 - Dshell Runner
File 'bin/product_configuration.md':
TimeoutException: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/byexample/common.py", line 172, in human_exceptions
    yield o
  File "/usr/local/lib/python2.7/dist-packages/byexample/byexample.py", line 16, in execute_examples
    return executor.execute(examples, filename)
  File "/usr/local/lib/python2.7/dist-packages/byexample/executor.py", line 46, in execute
    self.initialize_runners(runners, options)
  File "/usr/local/lib/python2.7/dist-packages/byexample/executor.py", line 22, in initialize_runners
    runner.initialize(options)
  File "/home/snoopy/noop/test/bextensions/dshell.py", line 96, in initialize
    self._exec_and_wait('sudo su admin\n', timeout=10)
  File "/usr/local/lib/python2.7/dist-packages/byexample/runner.py", line 193, in _exec_and_wait
    self._expect_prompt(timeout)
  File "/usr/local/lib/python2.7/dist-packages/byexample/runner.py", line 226, in _expect_prompt
    out)
TimeoutException: Prompt not found: the code is taking too long to finish or there is a syntax error.
Last 1000 bytes read:

**byexample version**
byexample 7.4.5

Note: 
The example is not the problem, because when I run it on a certain remote machine, it happens correctly.
But when I run it in one that is with the operating system without configuring (IP, gateway, etc) I throw that error. Precisely, the example would be responsible for assigning an ip, a gateway, and the other configurations.

Can it be because having no ip set, can not be accessed or something? A greeting !

**Comment by mbazan23** [comment]
product_configuration.md

## Configure a MC or a Sensor





Let's see what interface the product had detected automatically as its
management interface.

# Configure MC step

### Set the initial config items that duham normally does.

```shell
dshell> /config/system/primary_management_interface                             # byexample: +paste
<iface> (String)
```

Now we can configure the networking:
```shell
dshell> /config/system/interfaces/<iface>/ipv4 = <ip>                           # byexample: +paste
<ip> (String)
dshell> /config/system/interfaces/<iface>/ipv4_netmask = <netmask>              # byexample: +paste
<netmask> (String)
dshell> /config/system/ipv4_gateway = <gw>                                      # byexample: +paste
<gw> (String)
dshell> /config/system/dns_servers = [<resolv1>]                                # byexample: +paste
["<resolv1>"] (Array of String)
dshell> commit finish                                                           # byexample: +timeout=10
```

And finally, we enable the *read only user* (??):
```shell
dshell> /config/management_console/reporting_db/readonly_user_enabled = true
true (TrueClass)
```

Sniffing (DPI) tunning:
```shell
dshell> /config/performance/dpi_threads = <dpi-threats>                         # byexample: +paste
1 (Fixnum)
```


# Set the name of the node:

```shell
dshell> /config/system/node = <node-name>                                       # byexample: +paste +pass
```

## Register MC

```shell
dshell> registration start                                                      # byexample: +pass
dshell> registration status                                                     # byexample: +pass
nodename:<hostname>-nn model:<model> status:registered
dshell> registration checklist
Node Name<...>ok
Hostname<...>ok
Management IPv4<...>ok
Subnet Mask<...>ok
Default Gateway<...>ok
DNS servers<...>ok
```

## Configure SMTP and NTP hostnames; set the summary interval

```shell
dshell> /config/system/smtp/hostname = <smtp-hostname>                          # byexample: +paste
<smtp-hostname> (String)
dshell> /config/system/smtp/helo_name = <smtp-hostname>                         # byexample: +paste
<smtp-hostname> (String)
dshell> /config/system/ntp_server = <ntp-server>                                # byexample: +paste
<ntp-server> (String)
```

CSP only
dshell> /config/global/summary_interval = 300
300 (Fixnum)

Disable data sharing

```shell
dshell> /config/global/data_sharing = false
false (FalseClass)
```

Set the admin password to fspassword (check this, import harness is necessary)!!

```ruby
>> remote_commands_on('<host-name>', ['user password admin', 'fspassword', 'fspassword'], true)   # byexample: +paste +pass

```

**Comment by eldipa** [comment]
@mbazan23 hi!, first at all, thanks to taking some time to file a bug, it is always welcome the feedback of the community.

I changed/edited a part of your first comment and delete the second one as they are not relevant to the discussion of this issue. I hope I have not offended you.

Having saying that, let's talk about the issue.

As you said, the examples are not the problem (the file ``bin/product_configuration.md``). Another file with just one example of ``dshell`` should trigger the same issue.

You mentioned that the problem could be that the target vm has no IP, and it makes sense. If ``dshell`` establish a connection to a remote vm and this one has no IP, the ``dshell runner`` will fail.

However, it is possible that the problem is somewhere else: you said that the examples configure the IP but that does not mean that the vm had not one before. May be the vm got an IP from a DHCP.

If that's the case there could be other causes:
 - a firewall started to drop your packets and the ``dshell`` connection timedout.
 - the command ``sudo su admin`` may require a password and hang the rest of the execution. 

Try to upgrade to ``byexample 7.4.6`` (do a ``pip install -U byexample``). I backported a fix from the pre-release ``8.0.0`` that may improve the log on a timeout. That may give you a clue of what is happening.

Tell me how that resulted and thanks again for your time.
